tests/test_start_date.py

```diff
@@ -112,27 +118,66 @@ def test_run(self):
+
+                # verify all data from both syncs >= start_date
+                first_sync_target_mark = first_min_bookmarks.get(stream, {"mark": None})
+                second_sync_target_mark = second_min_bookmarks.get(stream, {"mark": None})
+
+                # get start dates for both syncs
+                first_sync_start_date = self.get_properties()["start_date"]
+                second_sync_start_date = self.start_date
+
+                # loop over minimum bookmark, the start date/state file date and replication key records for each
+                # syncs to verify the minimum bookmark is greater then or equal to start date/state file date
+                for start_date, target_mark, record_replication_keys in zip(
+                    (first_sync_start_date, second_sync_start_date),
+                    (first_sync_target_mark, second_sync_target_mark),
+                    (replication_key_sync_1, replication_key_sync_2)):
+                    target_value = next(iter(target_mark.values()))  # there should be only one
+
+                    if target_value:
 
-                if target_value:
+                        # it's okay if there isn't target data for a stream
+                        try:
+                            target_value = self.local_to_utc(parse(target_value))
 
-                    # it's okay if there isn't target data for a stream
-                    try:
-                        target_value = self.local_to_utc(parse(target_value))
+                            # verify that the minimum bookmark sent to the target for the second sync
+                            # is greater than or equal to the start date
+                            self.assertGreaterEqual(target_value,
+                                                    self.local_to_utc(parse(start_date)))
 
-                        # verify that the minimum bookmark sent to the target for the second sync
-                        # is greater than or equal to the start date
-                        self.assertGreaterEqual(target_value,
-                                                self.local_to_utc(parse(self.start_date)))
+                        except (OverflowError, ValueError, TypeError):
+                            print("bookmarks cannot be converted to dates, "
+                                "can't test start_date for {}".format(stream))
 
-                    except (OverflowError, ValueError, TypeError):
-                        print("bookmarks cannot be converted to dates, "
-                              "can't test start_date for {}".format(stream))
+                    # loop over every replication key records and verify we have
+                    # synced records greater than start date/state file date
+                    for record_replication_key in record_replication_keys:
+                        self.assertGreaterEqual(record_replication_key, start_date)
```

Comment:
I think the try except and assertion above are unnecessary. The minimum replication key value is covered by this assertion.